aptible/resource_log_drain.go

```diff
@@ -0,0 +1,241 @@
+package aptible
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/aptible/go-deploy/aptible"
+	strfmt "github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceLogDrain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLogDrainCreate,
+		Read:   resourceLogDrainRead,
+		Delete: resourceLogDrainDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceLogDrainImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"log_drain_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"env_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"handle": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"drain_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"database_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"drain_username": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"drain_host": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"drain_password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true, // The API generates a password if one isn't provided
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"drain_port": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"logging_token": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"drain_apps": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+			"drain_databases": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+			"drain_ephemeral_sessions": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+			"drain_proxies": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			// aliases
+			"token": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"pipeline": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"tags": {
```

Comment:
maybe same with tags? https://github.com/aptible/terraform-provider-aptible/pull/48/files#diff-3b8f370e03c85b07508c15ebb27706ec9cd341de8bd63367aa5ad957bbce47feR154

it looks like it uses a token possibly or checks on it? i'm not sure. disregard if not

Reply:
nvm looks insensitive disregard